src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -870,6 +873,58 @@ public S1(string s1, string s2) : this(s1)
                 // (15,12): warning CS8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
                 //     public S1(object obj1, object obj2) : this() // 4
                 Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S1").WithArguments("property", "Prop").WithLocation(15, 12));
+
+            var verifier = CompileAndVerify(source, options: WithNullableEnable());
+            verifier.VerifyDiagnostics(
+                // (7,9): warning CS8602: Dereference of a possibly null reference.
+                //         Prop.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop").WithLocation(7, 9),
+                // (12,9): warning CS8602: Dereference of a possibly null reference.
+                //         Prop.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop").WithLocation(12, 9),
+                // (15,12): warning CS8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public S1(object obj1, object obj2) : this() // 4
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S1").WithArguments("property", "Prop").WithLocation(15, 12));
+
+            verifier.VerifyIL("S1..ctor(string)", @"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldnull
+  IL_0002:  stfld      ""string S1.<Prop>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  call       ""readonly string S1.Prop.get""
+  IL_000d:  callvirt   ""string object.ToString()""
+  IL_0012:  pop
+  IL_0013:  ret
+}
+");
+
+            verifier.VerifyIL("S1..ctor(object, object)", @"
+{
+  // Code size        8 (0x8)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  initobj    ""S1""
+  IL_0007:  ret
+}
+");
+
+            verifier.VerifyIL("S1..ctor(string, string)", @"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""S1..ctor(string)""
+  IL_0007:  ldarg.0
+  IL_0008:  call       ""readonly string S1.Prop.get""
+  IL_000d:  callvirt   ""string object.ToString()""
+  IL_0012:  pop
+  IL_0013:  ret
+}
+");
         }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing nullable state of nested fields.
```csharp
#nullable enable
struct S1
{
    object F1;
    S2 S2;
    
    public S1()
    {
        F1.ToString();
        S2.F2.ToString();
    }
}
struct S2
{
    public object F2;
}
```

Reply:
Filed #60038. We are warnings in a comparable "baseline" scenario.